### ISSUE TITLE
Segmentation fault using `pdal info --all`

### ISSUE THREAD

**Issue by uclaros** [PDAL/PDAL#4264]
**Describe the bug**
Following command gives a segfault:

```
$ pdal info https://github.com/qgis/QGIS/raw/master/tests/testdata/point_clouds/copc/extrabytes-dataset.copc.laz --all
```

**Expected behavior**
Not segfault

**System/installation information:**

```
$ uname -a
Linux lutra-lutra 6.5.0-1-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.5.3-1 (2023-09-13) x86_64 GNU/Linux
```

```
$ pdal --version
--------------------------------------------------------------------------------------------------------------------------------------------------------------
pdal 2.6.0 (git-version: 823bc1)
--------------------------------------------------------------------------------------------------------------------------------------------------------------
```

**Comment by hobu** [comment]
```
==86444==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x0001122a9200 at pc 0x000107f3a578 bp 0x00016d035480 sp 0x00016d035478
READ of size 8 at 0x0001122a9200 thread T0
    #0 0x107f3a574 in std::__1::shared_ptr<pdal::MetadataNodeImpl>::operator->[abi:v160006]() const+0x44 (libpdalcpp.16.0.0.dylib:arm64+0x4f2574)
    #1 0x108770a78 in pdal::MetadataNode::children(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) const+0x1d0 (libpdalcpp.16.0.0.dylib:arm64+0xd28a78)
    #2 0x10876ff80 in pdal::MetadataNode::findChild(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) const+0x23c (libpdalcpp.16.0.0.dylib:arm64+0xd27f80)
    #3 0x108763820 in pdal::MetadataNode::findChild(char const*) const+0x160 (libpdalcpp.16.0.0.dylib:arm64+0xd1b820)
    #4 0x108ccc8c8 in pdal::stacProjection(pdal::MetadataNode&, pdal::MetadataNode&, pdal::MetadataNode&, pdal::MetadataNode&)+0x3d8 (libpdalcpp.16.0.0.dylib:arm64+0x12848c8)
    #5 0x108cc1678 in pdal::addStacMetadata(pdal::MetadataNode&, pdal::MetadataNode&, pdal::MetadataNode&, pdal::MetadataNode&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>)+0x1548 (libpdalcpp.16.0.0.dylib:arm64+0x1279678)
    #6 0x108cbe8e0 in pdal::InfoKernel::dump(pdal::MetadataNode&)+0x19c4 (libpdalcpp.16.0.0.dylib:arm64+0x12768e0)
    #7 0x108cbc1d4 in pdal::InfoKernel::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)+0xeec (libpdalcpp.16.0.0.dylib:arm64+0x12741d4)
    #8 0x108cc24b4 in pdal::InfoKernel::execute()+0x204 (libpdalcpp.16.0.0.dylib:arm64+0x127a4b4)
    #9 0x108df7b18 in pdal::Kernel::innerRun(pdal::ProgramArgs&)+0x440 (libpdalcpp.16.0.0.dylib:arm64+0x13afb18)
    #10 0x108df71a0 in pdal::Kernel::doExecution(pdal::ProgramArgs&)+0x20c (libpdalcpp.16.0.0.dylib:arm64+0x13af1a0)
    #11 0x108df6da4 in pdal::Kernel::run(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&, std::__1::shared_ptr<pdal::Log>&)+0x4a8 (libpdalcpp.16.0.0.dylib:arm64+0x13aeda4)
    #12 0x102dd4324 in App::execute(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>&, std::__1::shared_ptr<pdal::Log>&)+0xa04 (pdal:arm64+0x100010324)
    #13 0x102dd35c0 in main+0x360 (pdal:arm64+0x10000f5c0)
    #14 0x189d150dc  (<unknown module>)

0x0001122a9200 is located 0 bytes after 16-byte region [0x0001122a91f0,0x0001122a9200)
allocated by thread T0 here:
    #0 0x1038b10ec in wrap__Znwm+0x74 (libclang_rt.asan_osx_dynamic.dylib:arm64e+0x610ec)
    #1 0x107a63c98 in void* std::__1::__libcpp_operator_new[abi:v160006]<unsigned long>(unsigned long)+0x14 (libpdalcpp.16.0.0.dylib:arm64+0x1bc98)
    #2 0x107a63c3c in std::__1::__libcpp_allocate[abi:v160006](unsigned long, unsigned long)+0x44 (libpdalcpp.16.0.0.dylib:arm64+0x1bc3c)
    #3 0x108775a90 in std::__1::allocator<pdal::MetadataNode>::allocate[abi:v160006](unsigned long)+0x4c (libpdalcpp.16.0.0.dylib:arm64+0xd2da90)
    #4 0x1087757d4 in std::__1::__allocation_result<std::__1::allocator_traits<std::__1::allocator<pdal::MetadataNode>>::pointer> std::__1::__allocate_at_least[abi:v160006]<std::__1::allocator<pdal::MetadataNode>>(std::__1::allocator<pdal::MetadataNode>&, unsigned long)+0x124 (libpdalcpp.16.0.0.dylib:arm64+0xd2d7d4)
    #5 0x108775378 in std::__1::__split_buffer<pdal::MetadataNode, std::__1::allocator<pdal::MetadataNode>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<pdal::MetadataNode>&)+0x224 (libpdalcpp.16.0.0.dylib:arm64+0xd2d378)
    #6 0x1087747ec in std::__1::__split_buffer<pdal::MetadataNode, std::__1::allocator<pdal::MetadataNode>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<pdal::MetadataNode>&)+0x30 (libpdalcpp.16.0.0.dylib:arm64+0xd2c7ec)
    #7 0x1087737c8 in void std::__1::vector<pdal::MetadataNode, std::__1::allocator<pdal::MetadataNode>>::__push_back_slow_path<pdal::MetadataNode>(pdal::MetadataNode&&)+0x1a0 (libpdalcpp.16.0.0.dylib:arm64+0xd2b7c8)
    #8 0x108771b28 in std::__1::vector<pdal::MetadataNode, std::__1::allocator<pdal::MetadataNode>>::push_back[abi:v160006](pdal::MetadataNode&&)+0xcc (libpdalcpp.16.0.0.dylib:arm64+0xd29b28)
    #9 0x1087b2b7c in pdal::MetadataNode::children() const+0x42c (libpdalcpp.16.0.0.dylib:arm64+0xd6ab7c)
    #10 0x108ccc814 in pdal::stacProjection(pdal::MetadataNode&, pdal::MetadataNode&, pdal::MetadataNode&, pdal::MetadataNode&)+0x324 (libpdalcpp.16.0.0.dylib:arm64+0x1284814)
    #11 0x108cc1678 in pdal::addStacMetadata(pdal::MetadataNode&, pdal::MetadataNode&, pdal::MetadataNode&, pdal::MetadataNode&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>)+0x1548 (libpdalcpp.16.0.0.dylib:arm64+0x1279678)
    #12 0x108cbe8e0 in pdal::InfoKernel::dump(pdal::MetadataNode&)+0x19c4 (libpdalcpp.16.0.0.dylib:arm64+0x12768e0)
    #13 0x108cbc1d4 in pdal::InfoKernel::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)+0xeec (libpdalcpp.16.0.0.dylib:arm64+0x12741d4)
    #14 0x108cc24b4 in pdal::InfoKernel::execute()+0x204 (libpdalcpp.16.0.0.dylib:arm64+0x127a4b4)
    #15 0x108df7b18 in pdal::Kernel::innerRun(pdal::ProgramArgs&)+0x440 (libpdalcpp.16.0.0.dylib:arm64+0x13afb18)
    #16 0x108df71a0 in pdal::Kernel::doExecution(pdal::ProgramArgs&)+0x20c (libpdalcpp.16.0.0.dylib:arm64+0x13af1a0)
    #17 0x108df6da4 in pdal::Kernel::run(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&, std::__1::shared_ptr<pdal::Log>&)+0x4a8 (libpdalcpp.16.0.0.dylib:arm64+0x13aeda4)
    #18 0x102dd4324 in App::execute(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>&, std::__1::shared_ptr<pdal::Log>&)+0xa04 (pdal:arm64+0x100010324)
    #19 0x102dd35c0 in main+0x360 (pdal:arm64+0x10000f5c0)
    #20 0x189d150dc  (<unknown module>)
```

**Comment by hobu** [comment]
Addressed in #4266
